server/controllers/documents/generated/createGeneratedDocument.ts

```diff
@@ -31,9 +31,9 @@ export const createGeneratedDocument = async (req: Request, res: Response, next:
       const { documentId } = await generateRecallDocument(recallId, { category: cat, fileName }, token)
       res.locals.documentId = documentId
     }
+    next()
   } catch (err) {
     logger.error(err)
-  } finally {
-    next()
+    res.sendStatus(500)
```

Comment:
how is this displayed to end-user? Presumably we'd like them to try again?

Reply:
I looked into this but right now it just doesn't download it. We could certainly improve the UX, lets discuss with product. Maybe worth a spike. It's not straightforward as far as I can see, to issue a redirect on the download request and have it actioned by the browser.

Reply:
Cool, makes sense. I guess an error on the page would be ideal, is the 'download action' also doing the generate call as well? so its not that we can return the generate error to the UI before we 'open' the download.